node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1401,29 +1395,48 @@ impl BlockSignaturesCollector {
         }
         self.add(signatures);
 
-        if self.check_if_sufficient(validator_weights, ctx.config.finality_threshold_fraction()) {
+        if are_signatures_sufficient_for_sync_to_genesis(
+            self.check_if_sufficient(&validator_weights, ctx.config.finality_threshold_fraction()),
+        )
```

Comment:
Instead of wrapping `check_if_sufficient` in `are_signatures_sufficient_for_sync_to_genesis`, should we just call `are_...` inside `check_if_sufficient`? I think that would make more sense than wrapping every call site of `check_if_sufficient` in it (and I don't think there's any place here where we _want_ to fail on `TooManySignatures`?).

Reply:
Yup, that's a leftover from the initial approach where were failing (incorrectly) on `TooManySignatures` in some cases. Updated in 060b1f2e2819338337e9ae6c72680c250facff7e